recipes/wt/all/test_package/conanfile.py

```diff
@@ -1,19 +1,27 @@
-from conans import ConanFile, CMake, tools
-from conan.tools.build import cross_building
+from conan import ConanFile
+from conan.tools.build import can_run
+from conan.tools.cmake import cmake_layout, CMake
 import os
 
 
 class TestPackageConan(ConanFile):
     settings = "os", "arch", "compiler", "build_type"
-    generators = "cmake", "cmake_find_package_multi"
+    generators = "CMakeDeps", "CMakeToolchain", "VirtualRunEnv"
+    test_type = "explicit"
+
+    def requirements(self):
+        self.requires(self.tested_reference_str)
+
+    def layout(self):
+        cmake_layout(self)
 
     def build(self):
         cmake = CMake(self)
         cmake.configure()
         cmake.build()
 
     def test(self):
-        if not cross_building(self):
-            bin_path = os.path.join("bin", "test_package")
+        if can_run(self):
+            bin_path = os.path.join(self.cpp.build.bindirs[0], "test_package")
             args = " --docroot . --http-listen http://127.0.0.1:8080"
-            self.run(bin_path + args, run_environment=True)
+            self.run(bin_path + args, env="conanrun")
```

Comment:
```suggestion
            self.run(bin_path, env="conanrun")
```
No longer need arguments